src/ride_build.cpp

```diff
@@ -168,67 +175,96 @@ void RideBuildWindow::OnClick(WidgetNumber wid_num, const Point16 &pos)
 }
 
 /**
- * Checks whether the air space above the ground at the given location is suited to place a shop of the given height.
+ * Checks whether the air space above the ground at the given location is suited to place a fixed ride of the given height.
  * @param position Coordinate of the base voxel.
  * @param height Height of the object.
- * @return The space is suited to build the shop.
+ * @return The space is suited to build the ride.
  */
 static bool CheckSufficientVerticalSpace(const XYZPoint16& position, const int8 height)
 {
 	for (int8 h = 0; h <= height; ++h) {
 		const Voxel *v = _world.GetVoxel(position + XYZPoint16(0, 0, h));
-		if (v != nullptr && !v->CanPlaceInstance()) return false;
+		if (v == nullptr) continue;
+		if (!v->CanPlaceInstance() || v->GetGroundSlope() != SL_FLAT) return false;
+		if (h > 0 && v->GetGroundType() != GTP_INVALID) return false;
```

Comment:
Bit weird to first query the ground slope and if flat, only then ask whether there is any ground.

You may be lucky that non-existing ground is always flat, but I wouldn't rely on it. Please first check if there is valid ground before you discuss slopes of it.

An empty line after line 187 makes more clear that there are 2 different kinds of checks here. (Give readers visible clues that you change the topic, so even if one is  skimming the code it is less likely to miss the difference between "continue" and "return"